timelock-server/src/test/java/com/palantir/atlasdb/timelock/lock/AsyncLockServiceEteTest.java

```diff
@@ -276,14 +275,26 @@ public void timedOutRequestDoesNotHoldLocks() {
     @Test
     public void outstandingRequestsReceiveNotCurrentLeaderExceptionOnClose() {
         lockSynchronously(REQUEST_1, LOCK_A);
-        AsyncResult<LockToken> request2 = lock(REQUEST_2, LOCK_A);
+        AsyncResult<LeasableLockToken> request2 = lock(REQUEST_2, LOCK_A);
 
         service.close();
 
         assertThat(request2.isFailed()).isTrue();
         assertThat(request2.getError()).isInstanceOf(NotCurrentLeaderException.class);
     }
 
+    @Test
+    public void leaseShouldExpireBeforeReapingLocks() {
+        LeasableLockToken result = lock(REQUEST_1, LOCK_A).get();
+        assertThat(result.lease().isValid(service.identifiedTime())).isTrue();
+
+        waitForTimeout(TimeLimit.of(
```

Comment:
in most tests; recommend stubbing out time rather than sleeping - it makes the tests unnecessarily long-running